File: docs-examples/example-java/src/test/java/io/micronaut/configuration/openapi/docs/AccessorsStyleTest.java
package io.micronaut.configuration.openapi.docs;

import io.swagger.v3.oas.models.media.Schema;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class AccessorsStyleTest {

    @Test
    void customAccessorsAreDetected() {
        Schema<?> person = OpenApiSpec.load().getComponents().getSchemas().get("Person");

        assertEquals(List.of("name", "debtValue", "totalGoals"), List.copyOf(person.getProperties().keySet()));
        assertEquals("string", person.getProperties().get("name").getType());
        assertEquals("integer", person.getProperties().get("debtValue").getType());
        assertEquals("integer", person.getProperties().get("totalGoals").getType());
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/configuration/openapi/docs/AccessorsStyleTest.kt
package io.micronaut.configuration.openapi.docs

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class AccessorsStyleTest {

    @Test
    fun customAccessorsAreDetected() {
        val person = OpenApiSpec.load().components.schemas["Person"]!!

        assertEquals(listOf("name", "debtValue", "totalGoals"), person.properties.keys.toList())
        assertEquals("string", person.properties["name"]!!.type)
        assertEquals("integer", person.properties["debtValue"]!!.type)
        assertEquals("integer", person.properties["totalGoals"]!!.type)
    }
}
